docs/03-move/100-move-prover/02-mvp-tutorial.md

```diff
@@ -1,5 +1,574 @@
+
+```rust
+fun withdraw<CoinType>(addr: address, amount: u64) : Coin<CoinType> acquires Balance
+```
+
+Its role is to withdraw the `amount` of coins from the address `addr` and return it.
+
+### Specify the abort condition for `widthdraw`
+
+There are two possibilities for `withdraw` to abort:
+
+1. No resource of type `Balance<CoinType>` in `addr`.
+2. The balance in `addr` is less than `amount`.
+
+From these, we can define the abort condition like this:
+
+```rust
+spec withdraw {
+    let balance = global<Balance<CoinType>>(addr).coin.value;
+    aborts_if !exists<Balance<CoinType>>(addr);
+    aborts_if balance < amount;
+}
+```
+
+- A spec block can contain let bindings, which bind a long expression with a name that can be used repeatedly.
+`global<T>(addr): T` is a built-in function that returns a resource of type `T` at address `addr`.
+Here, we set balance to the number of tokens owned by addr via the let binding;
+- `exists<T>(address): bool` is a built-in function that returns true if resource `T` exists at address `addr`; otherwise returns false.
+
+The two lines of `aborts_if` statements correspond to the two conditions mentioned above.
+In general, if a function has multiple `aborts_if` conditions, the conditions are ORed together.
+
+As mentioned earlier, if we don't specify any abort conditions, Prover will not impose any restrictions on aborts.
+But once we give any kind of abort conditions, Prover defaults that we want to strictly check all possible aborts, so we need to list all possible conditions, which is equivalent to implicitly adding the instruction `pragma aborts_if_is_strict`.
+If only some of the conditions for abnormal exit are listed, the Prover will report a validation error.
+However, if the `pragma aborts_if_is_partial` is defined in the spec block, this is equivalent to telling the Prover:
+
+> I just want to list some of the conditions that will cause an abort, please just verify that it will abort under those conditions.
+
+If you are interested, you can do such a set of experiments to verify:
+
+- When deleting any of the above two `aborts_if` conditions, Prover will report an error;
+- When all `aborts_if` conditions are deleted at the same time, Prover will not report an error;
+- When adding `pragma aborts_if_is_partial`, no matter how many `aborts_if` conditions are kept, Prover will not report an error (of course, the conditions themselves must be correct).
+
+Some readers may be curious about the order of the three statements in the spec block:
+Why the definition of balance can be written after `aborts_if !exists<Balance<CoinType>>(addr)`.
+Because, if the latter holds true, `balance` does not actually exist.
+Wouldn't this order cause the Prover to fail?
+Simply put: no, the statements in the spec block are declarative and the order doesn't matter.
+
+For a more detailed understanding, you can refer to the MSL documentation for more information.
+
+### 指定 `withdraw` 的功能性质
```

Comment:
do you plan to leave this part as is and translate it later?

Reply:
I translated everything.